File: src/main/java/com/java/scogen/configprovider/SparkSessionProvider.java
package com.java.scogen.configprovider;

import org.apache.spark.sql.SparkSession;

import java.io.File;

public class SparkSessionProvider {

    private SparkSession spark;

    public SparkSession getSpark() {
        return spark;
    }

    public SparkSessionProvider(boolean hive)
    {
        if(hive) {
            String warehouseLocation = new File("spark-warehouse").getAbsolutePath();
            this.spark = SparkSession
                    .builder()
                    .master("local")
                    .appName("HiveTest")
                    .config("spark.sql.warehouse.dir", warehouseLocation)
                    .enableHiveSupport()
                    .getOrCreate();
        }else
            this.spark = SparkSession.builder().master("local").appName("HiveTest")
                    .getOrCreate();
    }
}

File: src/main/scala/com/scala/scogen/configprovider/SparkSessionProvider.scala
package com.scala.scogen.configprovider

import java.io.File

import org.apache.spark.sql.{Dataset, Row, SparkSession}

class SparkSessionProvider {

  val spark: SparkSession = SparkSession
    .builder.master("local")
    .appName("HiveTest")
    .getOrCreate

  val ds  = spark.read.csv("")
  spark.sparkContext.broadcast(ds);
}
